src/mono/wasm/runtime-test.js

```diff
@@ -91,138 +78,192 @@ if (typeof performance == 'undefined') {
-working_dir='/';
-while (args !== undefined && args.length > 0) {
-	if (args [0].startsWith ("--profile=")) {
-		var arg = args [0].substring ("--profile=".length);
+console.log("Arguments: " + testArguments);
+let profilers = [];
+let setenv = {};
+let runtime_args = [];
+let enable_gc = true;
+let working_dir='/';
+while (testArguments !== undefined && testArguments.length > 0) {
+	if (testArguments [0].startsWith ("--profile=")) {
+		const arg = testArguments [0].substring ("--profile=".length);
 
 		profilers.push (arg);
 
-		args = args.slice (1);
-	} else if (args [0].startsWith ("--setenv=")) {
-		var arg = args [0].substring ("--setenv=".length);
-		var parts = arg.split ('=');
+		testArguments = testArguments.slice (1);
+	} else if (testArguments [0].startsWith ("--setenv=")) {
+		const arg = testArguments [0].substring ("--setenv=".length);
+		const parts = arg.split ('=');
 		if (parts.length != 2)
-			fail_exec ("Error: malformed argument: '" + args [0]);
+			fail_exec ("Error: malformed argument: '" + testArguments [0]);
 		setenv [parts [0]] = parts [1];
-		args = args.slice (1);
-	} else if (args [0].startsWith ("--runtime-arg=")) {
-		var arg = args [0].substring ("--runtime-arg=".length);
-		runtime_args.push (arg);
-		args = args.slice (1);
-	} else if (args [0] == "--disable-on-demand-gc") {
+		testArguments = testArguments.slice (1);
+	} else if (testArguments [0].startsWith ("--runtime-arg=")) {
+		const arg = testArguments [0].substring ("--runtime-arg=".length);
+		runtime_args = testArguments.push (arg);
+		testArguments = testArguments.slice (1);
+	} else if (testArguments [0] == "--disable-on-demand-gc") {
 		enable_gc = false;
-		args = args.slice (1);
-	} else if (args [0].startsWith ("--working-dir=")) {
-		var arg = args [0].substring ("--working-dir=".length);
+		testArguments = testArguments.slice (1);
+	} else if (testArguments [0].startsWith ("--working-dir=")) {
+		const arg = testArguments [0].substring ("--working-dir=".length);
 		working_dir = arg;
-		args = args.slice (1);
+		testArguments = testArguments.slice (1);
 	} else {
 		break;
 	}
 }
-testArguments = args;
 
 // cheap way to let the testing infrastructure know we're running in a browser context (or not)
 setenv["IsBrowserDomSupported"] = is_browser.toString().toLowerCase();
 
-function writeContentToFile(content, path)
-{
-	var stream = FS.open(path, 'w+');
-	FS.write(stream, content, 0, content.length, 0);
-	FS.close(stream);
-}
-
-function loadScript (url)
-{
-	if (is_browser) {
-		var script = document.createElement ("script");
-		script.src = url;
-		document.head.appendChild (script);
-	} else {
-		load (url);
-	}
-}
-
-loadScript ("mono-config.js");
-
+// must be var as dotnet.js uses it
 var Module = {
 	mainScriptUrlOrBlob: "dotnet.js",
 
-	print,
-	printErr,
-
 	onAbort: function(x) {
-		print ("ABORT: " + x);
-		var err = new Error();
-		print ("Stacktrace: \n");
-		print (err.stack);
+		console.log ("ABORT: " + x);
```

Comment:
If we have console.error I would prefer that we use it here since this definitely calls for the special output treatment you get from console.error

Reply:
Agreed